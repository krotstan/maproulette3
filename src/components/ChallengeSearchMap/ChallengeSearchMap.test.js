import React from 'react'
import _cloneDeep from 'lodash/cloneDeep'
import { toLatLngBounds } from '../../services/MapBounds/MapBounds'
import { ChallengeLocation }
       from '../../services/Challenge/ChallengeLocation/ChallengeLocation'
import { ChallengeSearchMap } from './ChallengeSearchMap'

let basicProps = null

beforeEach(() => {
  basicProps = {
    mapBounds: {
      bounds: toLatLngBounds([0, 0, 0, 0]),
    },
    source: {id: 'foo'},
    visibleOverlays: [],
    updateChallengeSearchMapBounds: jest.fn(),
  }
})

test("it renders a full screen map", () => {
  const wrapper = shallow(
    <ChallengeSearchMap {...basicProps} />
  )

  expect(wrapper.find('.full-screen-map').exists()).toBe(true)

  expect(wrapper).toMatchSnapshot()
})

test("rerenders if the map bounds change is response to an external user action", () => {
  const wrapper = shallow(
    <ChallengeSearchMap {...basicProps} />
  )

  const newProps = _cloneDeep(basicProps)
  newProps.mapBounds.bounds = toLatLngBounds([-20, -20, 20, 20])
  newProps.mapBounds.fromUserAction = true

  expect(wrapper.instance().shouldComponentUpdate(newProps)).toBe(true)
})

test("rerenders if the default layer id changes", () => {
  const wrapper = shallow(
    <ChallengeSearchMap {...basicProps} />
  )

  const newProps = _cloneDeep(basicProps)
  newProps.source = {id: 'bar'}

  expect(wrapper.instance().shouldComponentUpdate(newProps)).toBe(true)
})

test("moving the map signals that the challenge search map bounds should be updated", () => {
  const bounds = [0, 0, 0, 0]
  const zoom = 3

  const wrapper = shallow(
    <ChallengeSearchMap {...basicProps} />
  )

  wrapper.instance().updateBounds(bounds, zoom, false)
  expect(basicProps.updateChallengeSearchMapBounds).toBeCalledWith(bounds, false)
})

test("moving the map signals that the challenges should be updated if filtering on map bounds", () => {
  basicProps.searchFilters = {location: ChallengeLocation.withinMapBounds}
  const bounds = [0, 0, 0, 0]
  const zoom = 3

  const wrapper = shallow(
    <ChallengeSearchMap {...basicProps} />
  )

  wrapper.instance().updateBounds(bounds, zoom, false)
  expect(basicProps.updateChallengeSearchMapBounds).toBeCalledWith(bounds, false)
})
