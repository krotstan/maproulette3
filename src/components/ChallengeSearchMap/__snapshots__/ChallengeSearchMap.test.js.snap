// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders a full screen map 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChallengeSearchMap
    mapBounds={
        Object {
            "bounds": Object {
              "_northEast": Object {
                "lat": 0,
                "lng": 0,
              },
              "_southWest": Object {
                "lat": 0,
                "lng": 0,
              },
            },
          }
    }
    source={
        Object {
            "id": "foo",
          }
    }
    updateChallengeSearchMapBounds={[Function]}
    visibleOverlays={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "ChallengeSearchMap",
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Connect(Connect(InjectIntl(_WithLayerSources))))
          mapBounds={
                    Object {
                              "bounds": Object {
                                "_northEast": Object {
                                  "lat": 0,
                                  "lng": 0,
                                },
                                "_southWest": Object {
                                  "lat": 0,
                                  "lng": 0,
                                },
                              },
                            }
          }
          source={
                    Object {
                              "id": "foo",
                            }
          }
          updateChallengeSearchMapBounds={[Function]}
          visibleOverlays={Array []}
/>,
        <EnhancedMap
          animate={true}
          animateFeatures={false}
          center={
                    Object {
                              "lat": 0,
                              "lng": 45,
                            }
          }
          initialBounds={
                    Object {
                              "_northEast": Object {
                                "lat": 0,
                                "lng": 0,
                              },
                              "_southWest": Object {
                                "lat": 0,
                                "lng": 0,
                              },
                            }
          }
          justFitFeatures={false}
          maxZoom={18}
          minZoom={2}
          onBoundsChange={[Function]}
          setInitialBounds={false}
          zoom={3}
          zoomControl={false}
>
          <ZoomControl
                    position="topright"
          />
          <InjectIntl(SourcedTileLayer)
                    mapBounds={
                              Object {
                                        "bounds": Object {
                                          "_northEast": Object {
                                            "lat": 0,
                                            "lng": 0,
                                          },
                                          "_southWest": Object {
                                            "lat": 0,
                                            "lng": 0,
                                          },
                                        },
                                      }
                    }
                    source={
                              Object {
                                        "id": "foo",
                                      }
                    }
                    updateChallengeSearchMapBounds={[Function]}
                    visibleOverlays={Array []}
                    zIndex={1}
          />
</EnhancedMap>,
        false,
      ],
      "className": "full-screen-map",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "mapBounds": Object {
            "bounds": Object {
              "_northEast": Object {
                "lat": 0,
                "lng": 0,
              },
              "_southWest": Object {
                "lat": 0,
                "lng": 0,
              },
            },
          },
          "source": Object {
            "id": "foo",
          },
          "updateChallengeSearchMapBounds": [Function],
          "visibleOverlays": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "animate": true,
          "animateFeatures": false,
          "center": Object {
            "lat": 0,
            "lng": 45,
          },
          "children": Array [
            <ZoomControl
              position="topright"
/>,
            <InjectIntl(SourcedTileLayer)
              mapBounds={
                            Object {
                                          "bounds": Object {
                                            "_northEast": Object {
                                              "lat": 0,
                                              "lng": 0,
                                            },
                                            "_southWest": Object {
                                              "lat": 0,
                                              "lng": 0,
                                            },
                                          },
                                        }
              }
              source={
                            Object {
                                          "id": "foo",
                                        }
              }
              updateChallengeSearchMapBounds={[Function]}
              visibleOverlays={Array []}
              zIndex={1}
/>,
            Array [],
            false,
          ],
          "initialBounds": Object {
            "_northEast": Object {
              "lat": 0,
              "lng": 0,
            },
            "_southWest": Object {
              "lat": 0,
              "lng": 0,
            },
          },
          "justFitFeatures": false,
          "maxZoom": 18,
          "minZoom": 2,
          "onBoundsChange": [Function],
          "setInitialBounds": false,
          "zoom": 3,
          "zoomControl": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "position": "topright",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "mapBounds": Object {
                "bounds": Object {
                  "_northEast": Object {
                    "lat": 0,
                    "lng": 0,
                  },
                  "_southWest": Object {
                    "lat": 0,
                    "lng": 0,
                  },
                },
              },
              "source": Object {
                "id": "foo",
              },
              "updateChallengeSearchMapBounds": [Function],
              "visibleOverlays": Array [],
              "zIndex": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          false,
        ],
        "type": [Function],
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "ChallengeSearchMap",
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Connect(Connect(InjectIntl(_WithLayerSources))))
            mapBounds={
                        Object {
                                    "bounds": Object {
                                      "_northEast": Object {
                                        "lat": 0,
                                        "lng": 0,
                                      },
                                      "_southWest": Object {
                                        "lat": 0,
                                        "lng": 0,
                                      },
                                    },
                                  }
            }
            source={
                        Object {
                                    "id": "foo",
                                  }
            }
            updateChallengeSearchMapBounds={[Function]}
            visibleOverlays={Array []}
/>,
          <EnhancedMap
            animate={true}
            animateFeatures={false}
            center={
                        Object {
                                    "lat": 0,
                                    "lng": 45,
                                  }
            }
            initialBounds={
                        Object {
                                    "_northEast": Object {
                                      "lat": 0,
                                      "lng": 0,
                                    },
                                    "_southWest": Object {
                                      "lat": 0,
                                      "lng": 0,
                                    },
                                  }
            }
            justFitFeatures={false}
            maxZoom={18}
            minZoom={2}
            onBoundsChange={[Function]}
            setInitialBounds={false}
            zoom={3}
            zoomControl={false}
>
            <ZoomControl
                        position="topright"
            />
            <InjectIntl(SourcedTileLayer)
                        mapBounds={
                                    Object {
                                                "bounds": Object {
                                                  "_northEast": Object {
                                                    "lat": 0,
                                                    "lng": 0,
                                                  },
                                                  "_southWest": Object {
                                                    "lat": 0,
                                                    "lng": 0,
                                                  },
                                                },
                                              }
                        }
                        source={
                                    Object {
                                                "id": "foo",
                                              }
                        }
                        updateChallengeSearchMapBounds={[Function]}
                        visibleOverlays={Array []}
                        zIndex={1}
            />
</EnhancedMap>,
          false,
        ],
        "className": "full-screen-map",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "mapBounds": Object {
              "bounds": Object {
                "_northEast": Object {
                  "lat": 0,
                  "lng": 0,
                },
                "_southWest": Object {
                  "lat": 0,
                  "lng": 0,
                },
              },
            },
            "source": Object {
              "id": "foo",
            },
            "updateChallengeSearchMapBounds": [Function],
            "visibleOverlays": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "animate": true,
            "animateFeatures": false,
            "center": Object {
              "lat": 0,
              "lng": 45,
            },
            "children": Array [
              <ZoomControl
                position="topright"
/>,
              <InjectIntl(SourcedTileLayer)
                mapBounds={
                                Object {
                                                "bounds": Object {
                                                  "_northEast": Object {
                                                    "lat": 0,
                                                    "lng": 0,
                                                  },
                                                  "_southWest": Object {
                                                    "lat": 0,
                                                    "lng": 0,
                                                  },
                                                },
                                              }
                }
                source={
                                Object {
                                                "id": "foo",
                                              }
                }
                updateChallengeSearchMapBounds={[Function]}
                visibleOverlays={Array []}
                zIndex={1}
/>,
              Array [],
              false,
            ],
            "initialBounds": Object {
              "_northEast": Object {
                "lat": 0,
                "lng": 0,
              },
              "_southWest": Object {
                "lat": 0,
                "lng": 0,
              },
            },
            "justFitFeatures": false,
            "maxZoom": 18,
            "minZoom": 2,
            "onBoundsChange": [Function],
            "setInitialBounds": false,
            "zoom": 3,
            "zoomControl": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "position": "topright",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "mapBounds": Object {
                  "bounds": Object {
                    "_northEast": Object {
                      "lat": 0,
                      "lng": 0,
                    },
                    "_southWest": Object {
                      "lat": 0,
                      "lng": 0,
                    },
                  },
                },
                "source": Object {
                  "id": "foo",
                },
                "updateChallengeSearchMapBounds": [Function],
                "visibleOverlays": Array [],
                "zIndex": 1,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            false,
          ],
          "type": [Function],
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
